content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+
+### Example 1
+In this example there's only one SRG, and no unbounded arrays yet. It is useful to understand register space & index binding assignment.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV1;      // Will be bound to t0 register.
+        Texture2D<float4>        m_texSRV2;      // Will be bound to t1 register.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u0 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u1 register.
+        Sampler                  m_sampler1;     // Will be bound to s0 register.
+        Sampler                  m_sampler2;     // Will be bound to s1 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
+```
+### Example 2: (Error case)
+This example is based on Example 1, but this time We are trying to introduce the first Unbounded Array: `Texture2D<float4> m_texSRV_unbounded[]`.  
+This example will fail to compile because, as mentioned before, an Unbounded Array takes over the whole register starting from their index of declaration.  
+In this example, `m_texSRV_unbounded`, is the first texture SRV so it will start at **t0**, and take over the rest of the indices until **tN**. Any other texture SRV declared afterwards won't find a register index to bind to.  
+```cpp
+    ShaderResourceGroupSemantic slot1
+    {
+        FrequencyId = 1;
+    };
+    
+    struct MyStruct
+    {
+        float3 m_a;
+        float3 m_b;
+    };
+    
+    //For DX12, --unique-idx is OFF, so the expected register assignment is as follows
+    ShaderResourceGroup SRG1 : slot1
+    {
+        Texture2D<float4>        m_texSRV_unbounded[]; // Will be bound to t0 register, and will take over t1+.
+        Texture2D<float4>        m_texSRV1;      // ERROR. There's no tN available to bind this resource to.
+        Texture2D<float4>        m_texSRV2;      // ERROR. There's no tN available to bind this resource to.
+        RWTexture2D<float4>      m_texUAV1;      // Will be bound to u0 register.
+    	RWTexture2D<float4>      m_texUAV2;      // Will be bound to u1 register.
+        Sampler                  m_sampler1;     // Will be bound to s0 register.
+        Sampler                  m_sampler2;     // Will be bound to s1 register.
+        ConstantBuffer<MyStruct> m_cb1; // Will be bound to b1 register. (Because The ConstantBuffer reserved for SRG1 will be bound to b0 register).
+        ConstantBuffer<MyStruct> m_cb2; // Will be bound to b2 register.
+    };
```

Comment:
```suggestion
ShaderResourceGroupSemantic slot1
{
    FrequencyId = 1;
};

struct MyStruct
{
    float3 m_a;
    float3 m_b;
};

//For DX12, --unique-idx is OFF, so the expected register assignment is as follows
ShaderResourceGroup SRG1 : slot1
{
    Texture2D<float4>        m_texSRV_unbounded[];  // Binds to t0 register, and takes over t1+.
    Texture2D<float4>        m_texSRV1;             // ERROR. There's no tN available to bind this resource to.
    Texture2D<float4>        m_texSRV2;             // ERROR. There's no tN available to bind this resource to.
    RWTexture2D<float4>      m_texUAV1;             // Binds to u0 register.
    RWTexture2D<float4>      m_texUAV2;             // Binds to u1 register.
    Sampler                  m_sampler1;            // Binds to s0 register.
    Sampler                  m_sampler2;            // Binds to s1 register.
    ConstantBuffer<MyStruct> m_cb1;                 // Binds to b1 register. (The ConstantBuffer reserved for SRG1 binds to b0 register).
    ConstantBuffer<MyStruct> m_cb2;                 // Binds to b2 register.
};
```
* Remove indent. It's not necessary when defining within a code snippet (```cpp). This also allows longer comments to fit on the page.
* Simplified "will be bound to" to "binds to".
* Simplified "will take over" to "takes over"

Reply:
Fixed.